apps/web/pages/profile/index.tsx

```diff
@@ -0,0 +1,336 @@
+/* eslint-disable react/jsx-props-no-spreading */
+/* eslint-disable react-hooks/rules-of-hooks */
+import React, { useState } from 'react';
+import type { NextPage } from 'next';
+import { signOut } from 'supertokens-auth-react/recipe/passwordless';
+import { ArrowBackIcon, CopyIcon } from '@chakra-ui/icons';
+import { Box, Button, Flex, Heading, Input, Stack, Text, useToast } from '@chakra-ui/react';
+import { useRouter } from 'next/router';
+import { useForm } from 'react-hook-form';
+import { Topbar, BottomBar } from '../../components/Navbar';
+
+type Form = {
+  campus: string | null;
+  desc: string | null;
+  district: string | null;
+  dob: string | null;
+  email: string | null;
+  house: string | null;
+  mentor: string | null;
+  mobile: string | null;
+  name: string | null;
+  pin: string | null;
+  pronoun: string | null;
+  skills: string | null;
+  street: string | null;
+};
+
+const index: NextPage = () => {
+  const router = useRouter();
+  const [edit, setEdit] = useState<boolean>(true);
+  const LogOut = async () => {
+    await signOut();
+    router.replace('/');
+  };
+  const {
+    register,
+    handleSubmit,
+    setFocus,
+    formState: { errors },
+  } = useForm<Form>();
+
+  const editHandler = () => {
+    setEdit(false);
+    // name input gets focused after pressing edit button
+    setFocus('name');
+  };
+
+  const toast = useToast();
+  const copyFile = () => {
+    toast({
+      title: 'Membership id copied.',
+      status: 'success',
+      duration: 1000,
+      isClosable: true,
+    });
+  };
+
+  const updateProfile = (data: Form) => {
+    if (
+      errors.campus ||
```

Comment:
U dont need to do this. React hook form can integrate validation like these and even advance with library like [yup](https://github.com/jquense/yup) and can be plugged with hook form

Ref: https://react-hook-form.com/get-started#SchemaValidation